src/compiler/performance.ts

```diff
@@ -101,22 +95,25 @@ namespace ts.performance {
      * @param cb The action to perform for each measure
      */
     export function forEachMeasure(cb: (measureName: string, duration: number) => void) {
-        measures.forEach((measure, key) => {
-            cb(key, measure);
+        perfEntryList?.getEntriesByType("measure").forEach(entry => {
+            cb(entry.name, entry.duration);
         });
     }
 
     /** Enables (and resets) performance measurements for the compiler. */
     export function enable() {
-        counts = new Map<string, number>();
-        marks = new Map<string, number>();
-        measures = new Map<string, number>();
-        enabled = true;
-        profilerStart = timestamp();
+        if (!enabled) {
+            perfHooks ||= tryGetNativePerformanceHooks() || ShimPerformance?.createPerformanceHooksShim(timestamp);

```

Comment:
Under what circumstances is the shim un/available?

Reply:
The shim is only loaded in the language service/tsserver. `tsc.ts` never loads the shim. That's the same behavior we use for the `Map` and `Set` shims.

Reply:
Interesting.  Does the LS use the perf code?

Reply:
Not currently, but in theory it could. Especially if we end up with a way to have these user timings appear in a trace, and wanted to be able to initiate a trace against the language service from within the editor. This is actually a fairly reasonable possibility, since VS Code does have this functionality (although, IIRC, it only currently profiles the "in-process" extension, not the language service):

![image](https://user-images.githubusercontent.com/3902892/93401806-ec536700-f837-11ea-9479-8ff61c4a90b4.png)
(from the `Developer: Show Running Extensions` command).

Reply:
Are the shims for Monaco or something?  I'm pretty sure both VS and VS Code are on node >= 10?

Reply:
The shim would be for any non-node Host, such as Monaco in the browser, or if anyone else runs TS in the browser. The `ts.performance` namespace is marked `/* @internal */` so its unlikely anyone is actually using it in those scenarios, however.

It doesn't look like you can trigger it on the playground with `// @diagnostics: true` (as far as I can tell, maybe @orta knows for sure). I'm happy to drop the shim if that's what everyone else wants.

Reply:
That will definitely set the compiler option in the TSServer in the playground:

![Screen Shot 2020-09-17 at 12 11 46 PM](https://user-images.githubusercontent.com/49038/93497699-b9b07980-f8de-11ea-825d-861e9b42fbc4.png)

From the looks of caniuse.com for this API, the last browser to add the API was Safari in 2017 - which _might_ fall into Monaco's support window. They basically say "any desktop browser but IE".

 It's outside  the TS website's support window though (I use this API at the [bottom of handbook pages](https://www.typescriptlang.org/docs/handbook/intro.html#site-footer) but skip showing it on "ancient" browsers) 

So, I don't really have a definitive answer I'm afraid